### ISSUE TITLE
Convert Factories from Eclipse Che website to point to che.openshift.io

### ISSUE THREAD

**Issue by l0rd** [redhat-developer/rh-che#709]
This is about:
1. creating a specific osio account for the factories (i.e. codereadyfactories@redhat.com)
2. create the 22 factories corresponding to the ([codenvy.io factories](https://www.eclipse.org/che/getting-started/cloud/))
3. if some factory doesn't work on osio create an issue (fixing the factory is not part of this issue) 
4. update the factory links on eclipse.org

**Comment by riuvshin** [comment]
hi, afaik this should be done before JUL 10, so I wanted to make sure it is in the current sprint.

cc: @kbsingh

**Comment by ibuziuk** [comment]
@riuvshin this issue is not in the current osio team sprint 
@l0rd do you think we should reconsider it ?

**Comment by l0rd** [comment]
@ibuziuk yes I have update the description to discuss about adding it to current sprint during our daily today. @riuvshin there are good chances that it will be part of current sprint.

**Comment by riuvshin** [comment]
hi guys! that's a good news! thanks a lot!

**Comment by l0rd** [comment]
@riuvshin @slemeur we have added this issue to current sprint

**Comment by slemeur** [comment]
Thanks @l0rd !

**Comment by benoitf** [comment]
FYI here is factory config for che-docs
https://gist.github.com/benoitf/0c7182bd7eb92005faf9b26789ddfeaf

you can try it live at https://che.openshift.io/f?id=factoryi9sc51dcf0vg6920

**Comment by davidfestal** [comment]
The Codenvy.io factories have been migrated thanks to this tool: http://che-factory-migration-to-osio-dfestal-run.8a09.starter-us-east-2.openshiftapps.com/ (source [here](https://github.com/davidfestal/che-factory-migration-to-osio))  developed on OSIO production.

The resulting factories have been migrated unto the newly-provisioned `codeready-factories` OSIO account.

I tested all the migrated factories and summarized the results, as well as the URLs to test them, in the following document: https://docs.google.com/spreadsheets/d/1JpA3uSjfneeAqeNcchwmI1T0sxpO_o_FTaTAiVl0IIw/edit?usp=sharing

In this Google Sheet document, you can use filtered views to filter :
- factories that fail due to *incompatible images*:
    add `&fvid=4415306` at the end of the doc URL 

- factories that start a workspace but then fail to execute command since it *requires sudo rights*:
    add `&fvid=1380154325` at the end of the doc URL

- factories start but seem to have a problem with Git cloning being very long and unfinished when the initial command starts:
    add `&fvid=218870918` at the end of the doc URL 

- factories start but seem to have a problem with when the initial command starts (unrelated to SUDO):
    add `&fvid=2049116502` at the end of the doc URL 

To get the Json file with the list of factories to replace in the Eclipse web, you can use the [migration tool](http://che-factory-migration-to-osio-dfestal-run.8a09.starter-us-east-2.openshiftapps.com/):
- enter the Che server API URL : `https://che.openshift.io/api/`
- enter the OSIO token of the `codeready-factories` OSIO user
- In the `Migration` tab, click on the `Show all migrated` button.

**Comment by riuvshin** [comment]
@davidfestal is https://www.eclipse.org/che/getting-started/cloud/ updated?

**Comment by davidfestal** [comment]
No. But I can do it if you tell me where to change it.

**Comment by slemeur** [comment]
@riuvshin : you too ;) every commiter on eclipse in fact!

**Comment by slemeur** [comment]
Before we do that, and replace the factories on che website, we need to make sure we have some telemetry events in place for logging:
- When a workspaces is started from a factory
- What factory has been started
- When an account is created on OSIO from a user who clicked on a factory.

Would we be able to collect that?

**Comment by riuvshin** [comment]
yeah I do remember that but I don't know where it is :) I meant you know how to do this:)

regarding telemetry, I don't know if that is a mandatory thing, if yes it can wait but as for me, we can enable that anytime later.  My only concern is that this part of the traffic should be switched already according to the plan :)

**Comment by ibuziuk** [comment]
@slemeur AFAIK, we do not have this telemetry events processing in place. What's more the last event -'When an account is created on OSIO from a user who clicked on a factory' would need to be processed not on che side, but on osio registration side. IMO we need a separate issue for processing this  telemetry events

**Comment by benoitf** [comment]
FYI some data were on https://github.com/codenvy-legacy/factories/tree/factory-2.0/che-website

**Comment by ibuziuk** [comment]
We are going to close this issue. Remained issue related to factory transition are
- https://github.com/redhat-developer/rh-che/issues/729
- https://github.com/redhat-developer/rh-che/issues/749

**Comment by riuvshin** [comment]
totally agree with @ibuziuk
